### ISSUE TITLE
Improve speed of Jenkins builds

### ISSUE THREAD

**Issue by oskarth** [status-im/status-react#1708]
## User Story

As a developer, I want builds to finish quickly so I can see if things work / QA can quickly test binaries.

## Summary
When there are new dependencies it can take over 40 minutes to build. It seems like this can be improved drastically with a better instance type. See http://185.90.37.243:8080/blue/organizations/jenkins/Github%20Org%20Builder%2Fstatus-react/detail/feature%2Fpush-notifications-spike-%231599/7/pipeline/

In addition:
-  we should setup DNS so we can go to `build.status.im` or something like that
- CocoaPods should be updated to 1.3.1

*Type*: Feature (tooling)

**Comment by oskarth** [comment]
Basic profile comparison of Jenkins and building iOS locally with mid-line MBP Pro 2017 for http://185.90.37.243:8080/blue/organizations/jenkins/Github%20Org%20Builder%2Fstatus-react/detail/feature%2Fpush-notifications-spike-2-%231599/7/pipeline/7

Total build: 2h 4m. Git&Deps 18m, Tests 5m, Build 5m, Build (Android) 58m, Deploy 0m, Build iOS 36m, Deploy 0m, Slack Notification 0m. Let's focus on three areas: Git&Deps, build iOS and ordering. Only tested for iOS, but it seems likely build iOS and build Android are similar in terms of bottlenecks and generic steps.

### 1) Git & Deps

Took 18m for Jenkins, takes 9m locally.

(a) Twice the time, which suggest CPU/RAM/more instances could cut this in half.

(b) This removes the `node_modules` directory, which I guess is the prudent thing to do, and it isn't a huge win in the grand scheme of thing. It might be worth revisiting though, as very often `node_modules` don't change.

(c) `npm install` is obviously the slow part here, and there's a particular slow part of this: `node ./scripts/download-mapbox-gl-native-ios-if-on-mac.js 3.3.1`. Locally this can take 5m+, or around 70% of the install time (locally) due to some crazy install procedure. It even times out sometimes. I think we can vendor this or figure out a better solution, but it might not be super trivial.

### 2) Build iOS

36m on Jenkins, 5m locally. Normally this step is quicker on Jenkins, so it seems the box got overwhelmed with other jobs, but that just illustrates the point. 7x difference is huge.

### 3) Ordering

Instead of building iOS and Android sequentially, we can consider building them as separate builds in parallel, or we can notify on slack with the most easily tested platform first, before moving on to the next one. This would build time in half.

### Summary

There are lots of things we can do to cut the building time in half many times over. Let's start with the easy ones. As a start, I suggest attaching to the box, running a few builds and check `htop`. I don't know the specifics of how Jenkins is setup wrt parallelization of builds, but that might be a separate piece of work.

**Comment by oskarth** [comment]
Notes after talking to @rasom 

We can see here http://185.90.37.243:8080/job/status-react-nightly/ (build #283) that if multiple jobs aren't running simultaneously Git&Deps takes 6-7m and building iOS 14m, which is in the ballpark of running things locally. The issue is thus with multiple builds running concurrently.

Running multiple macos instances might be expensive. Also need to configure server, if company that provides Jenkin servers can just copy instance that'd simplify things. Check in with @jarradh next week.

With multiple instances we could also potentially split up the Android and iOS builds, so the expensive macos instances don't need to impact Android builds. This would cut build time in half and allow for an arbitrary number of concurrent builds (on average we seem to have a few builds running at any given time, so this might be a factor of 2-3 on peak commit/test hours).

**Comment by oskarth** [comment]
As a first step: https://github.com/status-im/status-react/issues/1828

**Comment by oskarth** [comment]
Issue and profiling a bit outdated but many observations still stand. Closing this in favor of more targeted https://github.com/status-im/status-react/issues/2880
